pages/admin_pages.py

```diff
@@ -193,3 +199,46 @@ def reset_download_api_credentials():
         % (access_key, secret_key)
     flash(Markup(msg), 'warning')
     return redirect("/manage_credentials")
+
+
+@admin_pages.route('/new_api_key', methods=['POST'])
+@authenticate_researcher_login
+def new_api_key():
+    tableau_api_permission = 'api_permission' in request.values
+    readable_name = request.values.get("readable_name", "")
+    researcher = Researcher.objects.get(username=session[SESSION_NAME])
+    api_key = ApiKey.generate(researcher=researcher, has_tableau_api_permissions=tableau_api_permission, readable_name=readable_name)
+    msg = """<h3>New Data-Download API credentials have been generated for you!</h3>
+        <p>Your new <b>Access Key</b> is:
+          <div class="container-fluid">
+            <textarea rows="1" cols="85" readonly="readonly" onclick="this.focus();this.select()">%s</textarea></p>
+          </div>
+        <p>Your new <b>Secret Key</b> is:
+          <div class="container-fluid">
+            <textarea rows="1" cols="85" readonly="readonly" onclick="this.focus();this.select()">%s</textarea></p>
+          </div>
+        <p>Please record these somewhere; This secret key will not be shown again!</p>""" \
+          % (api_key.access_key_id, api_key.access_key_secret_plaintext)
+    api_key.save()
+    flash(Markup(msg), 'warning')
+    return redirect("/manage_credentials")
+
+
+@admin_pages.route('/disable_api_key', methods=['POST'])
+@authenticate_researcher_login
+def disable_api_key():
+    if "api_key_id" not in request.values:
+        flash(Markup("No API key specified"), 'warning')
+        return redirect("/manage_credentials")
+    api_key_id = request.values["api_key_id"]
+    api_key_query = ApiKey.objects.filter(access_key_id=api_key_id).filter(researcher=get_session_researcher())
+    if not api_key_query.exists():
+        flash(Markup("No matching API key found to disable"), 'warning')
+        return redirect("/manage_credentials")
+    api_key = api_key_query[0]
+    if not api_key.is_active:
+        flash("That API key has already been disabled: %s" % api_key_id, 'warning')
+        return redirect("/manage_credentials")
+    api_key.is_active = False
+    api_key.save()
```

Comment:
You could also collapse this whole thing into:
`ApiKey.objects.filter(researcher=get_session_researcher(), access_key_id=api_key_id, is_active=True).update(is_active=False)`